docs/site/src/components/GitHubCodeBlock/GitHubCodeBlock.tsx

```diff
@@ -1,93 +1,58 @@
-import React, { useEffect, useState } from 'react';
-import axios from 'axios';
-import CodeBlock from "@theme/CodeBlock";
 import { useColorMode } from '@docusaurus/theme-common';
+import CodeBlock from "@theme/CodeBlock";
+import { useEffect, useState } from 'react';
+import useGitHubCodeBlock from './useGitHubCodeBlock';
 
 import './GitHubCodeBlock.css';
 
 const GitHubCodeBlock = ({ repoUrl }) => {
-  const [code, setCode] = useState('');
-  const [language, setLanguage] = useState('plaintext');
-  const [sourceUrl, setSourceUrl] = useState('');
+  const [data, setData] = useState(null);
+  const colorMode = useColorMode();
 
   useEffect(() => {
-    async function fetchCode() {
-      // Use a single regex that can optionally capture line numbers
-      const regex = /(https:\/\/github\.com\/)([^\/]+)\/([^\/]+)\/blob\/([^\/]+)\/([^#]+)#L(\d+)-L(\d+)/;
-      var match = repoUrl.match(regex);
-      if (!match) {
-        const noLinesRegex = /(https:\/\/github\.com\/)([^\/]+)\/([^\/]+)\/blob\/([^\/]+)\/([^#]+)/;
-        match = repoUrl.match(noLinesRegex);
+    const fetchData = async () => {
+      try {
+        const result = await useGitHubCodeBlock(repoUrl);
+        setData(result);
+      } catch (error) {
+        console.error(error);
       }
+    };
```

Comment:
A hook is either a base react hook, or a function that uses a base react hook. In this case `useGithubCodeBlock`. It just fetches the data. 

I suggest moving `useEffect` and `useState` logic into `useGithubCodeBlock`, and using that as a hook.